rust/noosphere-core/src/data/address.rs

```diff
@@ -99,11 +183,37 @@ impl LinkRecord {
                 }
             }
         }
-
         warn!("No facts contained a link!");
-
         None
     }
+
+    /// Generates a [Capability] struct representing permissions in a [LinkRecord].
+    ///
+    /// ```
+    /// use noosphere_core::{data::LinkRecord, authority::{SphereAction, SphereReference}};
+    /// use ucan::capability::{Capability, Resource, With};
+    ///
+    /// let identity = "did:key:z6MkoE19WHXJzpLqkxbGP7uXdJX38sWZNUWwyjcuCmjhPpUP";
+    /// let expected_capability = Capability {
+    ///     with: With::Resource {
+    ///         kind: Resource::Scoped(SphereReference {
+    ///            did: identity.to_owned(),
+    ///         }),
+    ///     },
+    ///     can: SphereAction::Publish,
+    /// };
+    /// assert_eq!(LinkRecord::generate_capability(&identity), expected_capability);
+    /// ```
+    pub fn generate_capability(identity: &str) -> Capability<SphereReference, SphereAction> {
```

Comment:
A generic wrapper for capability generation might be more appropriate e.g., `generate_capability(sphere: &Did, action: SphereAction)`, and we can stick it in `noosphere-core::authority`.